src/components/templates/PosterPage/components/ShareModal/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { FACEBOOK_APP_ID } from "secrets";
+import { PosterPageVenue } from "types/venues";
+import { WithId } from "utils/id";
+
+const getCategoriesFromVenue = (venue: WithId<PosterPageVenue>) => {
+  return venue?.poster?.categories || [];
+};
+
+export const getTitleTextForSharing = (
+  venue: WithId<PosterPageVenue>,
+  url: string
+) => {
+  if (!venue.poster) return "Check out this OHBM Poster";
+
+  return `Check out this OHBM Poster, ${venue.poster.title} by ${venue.poster.authorName} at ${url}`;
+};
+
+export const getFacebookHref = (
+  venue: WithId<PosterPageVenue>,
+  url: string,
+  text: string
+) => {
+  const categories = getCategoriesFromVenue(venue);
+  const hashtagsString = categories.reduce((result, hash, index, array) => {
+    const isLast = index === array.length - 1;
+    const separator = isLast ? "" : ", ";
+    return result + `#${hash}${separator}`;
+  }, "");
+
+  return `https://www.facebook.com/sharer/sharer.php?app_id=${FACEBOOK_APP_ID}&u=${url}&quote=${text}&hashtag=${hashtagsString}`;
```

Comment:
[NB] We would probably be better to use the JS URL helpers for constructing this URL, rather than joining strings. I believe that will handle encoding/escaping for us automatically then.

- https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams
- https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams

Reply:
And these should be done using functions from `utils/url` (or create new appropriate functions to do so there)

Reply:
https://github.com/sparkletown/sparkle/pull/1396/commits/444fbee5a64755e19d21cc3020703b59dbd60b69